### PR TITLE
fix: resolve fee percentage calculation bugs and test data inconsistencies

### LINKED ISSUE #4447: Agreement total in agreements list not matching total on detail page
## Expected Behavior
The amount shown in the "Agreement Total" column of the agreements list, and in the agreements list export, should match the total of (non-draft?) budget lines found on the "Agreement Total" summary card on the SCs and budget lines tab of an agreement's details page. 

## Current Behavior
Currently, the amount shown for the agreement total on the agreements list page or the agreements export is wildly larger than the value shown in the card on the agreement details page. sometimes orders of magnitude larger.  It doesn't appear to be due to draft BLs or the corresponding project's sum of agreements. 

## Possible Cause
unknown

## Steps to Reproduce
1. Navigate to the agreement list. Note in the matrix the agreement total for agreement id 6, for example, using the sample dataset.  
2. Then view agreement 6's details. 
3. Click SCs & Budget lines tab. 
4. Note that value shown for agreement subtotal and total in the summary card.

## Context
seen in all environments. reported by OPRE.

## Possible Implementation
there should be tests to address this in place. please also test out exporting as well.

### PR DESCRIPTION
**Pull request by fpigeonjr**
## Summary

Fixed critical fee percentage calculation bug affecting agreement totals throughout the application

• Root cause: Components were manually dividing fee percentages by 100 before passing to calculateTotal, but the function expected decimal rates
• Impact: Agreement fees were being calculated as 1/100th of correct value (e.g., 5% fee showing as 0.05%)
• Resolution: Refactored calculateAgreementTotal to handle percentage-to-decimal conversion internally, removing all manual /100 divisions from component calls

### Core Function Refactoring

• Refactored calculateTotal → calculateAgreementTotal to divide by 100 internally
• Added calculateFeeTotal function for dedicated fee calculations
• Enhanced getProcurementShopFees with improved parameter handling
• Updated all function calls to pass whole number percentages (5.0 instead of 0.05)

### Bug Fixes (7 Components)

Removed incorrect /100 divisions from:

• ProcurementShopReviewCard.jsx (lines 45, 50)
• AgreementEditForm.jsx (lines 169, 172)
• AgreementBLIAccordion.jsx (lines 88-89)
• AgreementChangesResponseAlert.hooks.jsx (lines 121-122)
• CreateBLIsAndSCs.hooks.js (lines 410, 499)
• useChangeRequests.hooks.js (lines 81, 86)
• Agreement.jsx (lines 116, 120)

### Test Data Standardization (6 Test Files)

• Ensured consistent whole number fee percentage format (5.0) across all test files
• Added 22+ comprehensive test cases covering edge cases, typical rates, and error conditions

### Documentation & Prevention

• Updated AGENTS.md with clear fee percentage format convention
• Added explicit examples of correct usage patterns

Closes #4447 - Agreement totals now calculate correctly across all components

### CODE REVIEW

File: frontend/src/helpers/agreement.helpers.js [hunk 1]
```diff
@@ -33,14 +33,15 @@ export const getAgreementSubTotal = (agreement) => {
...
+ * @param {number} feeRate - The fee rate as a percentage (e.g., 5 for 5%).
+ * @param {boolean} isAfterApproval - Whether to include DRAFT budget lines or not.
  * @returns {number} The total cost of the items.
  */
 export const calculateTotal = (budgetLines, feeRate, isAfterApproval = false) => {
     return (
         budgetLines
             ?.filter(({ status }) => (isAfterApproval ? true : status !== BLI_STATUS.DRAFT))
-            .reduce((acc, { amount = 0 }) => acc + amount * feeRate, 0) || 0
+            .reduce((acc, { amount = 0 }) => acc + amount * (feeRate / 100), 0) || 0
```
Comment (fpigeonjr) [inline frontend/src/helpers/agreement.helpers.js:44]: this is the crux of the PR

**Comment by jonnalley** [comment]
Many thanks for digging into this and resolving so quickly.  A bit of a testing quality fail that we got here in the first place, but, we live and learn.

File: AGENTS.md [hunk 2]
```diff
@@ -221,3 +221,22 @@ The frontend follows modern React patterns with Redux for state management:
 - Database migrations should be reviewed before applying
 - RSA keys are required for JWT functionality in development
 - Use `pipenv shell` to avoid prefixing commands with `pipenv run`
+
+### Fee Percentage Format Convention
```
Comment (jonnalley) [inline AGENTS.md:225]: 🎉

**Review by Santi-3rd — APPROVED** [review]

File: frontend/src/helpers/agreement.helpers.js [hunk 3]
```diff
@@ -33,14 +33,44 @@ export const getAgreementSubTotal = (agreement) => {
 /**
  * Calculates the total cost of a list of items, taking into account a fee per item and non-DRAFT budgetlines.
  * @param {import("../types/BudgetLineTypes").BudgetLine[]} budgetLines - The list of items to calculate the total cost for.
- * @param {number} feeRate - The fee per item.like 0.005
+ * @param {number | null} feeRate - The fee rate as a percentage (e.g., 5 for 5%).
+ * @param {boolean} [isAfterApproval] - Whether to include DRAFT budget lines or not.
  * @returns {number} The total cost of the items.
  */
-export const calculateTotal = (budgetLines, feeRate, isAfterApproval = false) => {
+export const calculateAgreementTotal = (budgetLines, feeRate, isAfterApproval = false) => {
```
Comment (fpigeonjr) [inline frontend/src/helpers/agreement.helpers.js:40]: this is the crux of the PR
